pkg/customresourcestate/registry_factory.go

```diff
@@ -257,22 +244,55 @@ func (c *compiledGauge) Values(v interface{}) (result []eachValue, errs []error)
 
 type compiledInfo struct {
 	compiledCommon
+	labelFromKey string
 }
 
 func (c *compiledInfo) Values(v interface{}) (result []eachValue, errs []error) {
-	if vs, isArray := v.([]interface{}); isArray {
-		for _, obj := range vs {
+	onError := func(err ...error) {
+		errs = append(errs, fmt.Errorf("%s: %v", c.Path(), err))
+	}
+
+	switch iter := v.(type) {
+	case []interface{}:
+		for _, obj := range iter {
 			ev, err := c.values(obj)
 			if len(err) > 0 {
-				errs = append(errs, err...)
+				onError(err...)
 				continue
 			}
 			result = append(result, ev...)
 		}
-		return
+	case map[string]interface{}:
+		value, err := c.values(v)
+		if err != nil {
+			onError(err...)
+			break
+		}
+		for _, ev := range value {
+			if _, ok := ev.Labels[c.labelFromKey]; ok {
+				onError(fmt.Errorf("labelFromKey (%s) generated labels conflict with labelsFromPath, consider renaming it", c.labelFromKey))
+				continue
+			}
+		}
+		// labelFromKey logic
+		for key := range iter {
+			if key != "" && c.labelFromKey != "" {
+				result = append(result, eachValue{
+					Labels: map[string]string{
+						c.labelFromKey: key,
+					},
+					Value: 1,
+				})
+			}
+		}
+		if len(result) == 0 {
+			result = value
+		}
```

Comment:
This is what I was referring to in https://github.com/kubernetes/kube-state-metrics/pull/1880#discussion_r1017494200, we need value to be added to the list everytime no? Otherwise the result will not contain path labels when we already added some stuff to the list.
```suggestion
	        result = append(result, value)
```

Reply:
Makes sense, patched.